Zend/zend_object_handlers.c

```diff
@@ -1714,6 +1723,11 @@ ZEND_API int zend_std_compare_objects(zval *o1, zval *o2) /* {{{ */
 {
 	zend_object *zobj1, *zobj2;
 
+	if (zend_objects_check_stack_limit()) {
+		zend_throw_error(NULL, "Object compare - stack limit reached");
```

Comment:
Nit:

``` suggestion
		zend_throw_error(NULL, "Maximum call stack size reached during object comparison");
```

To get a similar message as in https://github.com/php/php-src/blob/05618e7e0b9f224161c729f09ec99af544f81543/Zend/zend_compile.c#L114